src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/UWPApplication.cs

```diff
@@ -394,72 +378,277 @@ internal string LogoPathFromUri(string uri, string theme)
+                ProgramLogger.LogException(
+                            $"|UWP|LogoPathFromUri|{Package.Location}" +
+                            $"|{UserModelId} can't find logo uri for {uri} in package location: {Package.Location}", new FileNotFoundException());
+            }
+        }
+
         public ImageSource Logo()
         {
-            var logo = ImageFromPath(LogoPath);
-            return logo;
+            if (LogoType == LogoType.Colored)
+            {
+                var logo = ImageFromPath(LogoPath);
+                var platedImage = PlatedImage(logo);
+                return platedImage;
+            }
+            else
+            {
+                return ImageFromPath(LogoPath);
+            }
+        }
+
+        private ImageSource PlatedImage(BitmapImage image)
+        {
+            if (!string.IsNullOrEmpty(BackgroundColor))
+            {
+                string currentBackgroundColor;
+                if (BackgroundColor == "transparent")
+                {
+                    currentBackgroundColor = SystemParameters.WindowGlassBrush.ToString(CultureInfo.InvariantCulture);
+                }
+                else
+                {
+                    currentBackgroundColor = BackgroundColor;
+                }
+
+                var padding = 8;
+                var width = image.Width + (2 * padding);
+                var height = image.Height + (2 * padding);
+                var x = 0;
+                var y = 0;
+
+                var group = new DrawingGroup();
+                var converted = ColorConverter.ConvertFromString(currentBackgroundColor);
+                if (converted != null)
+                {
+                    var color = (Color)converted;
+                    var brush = new SolidColorBrush(color);
+                    var pen = new Pen(brush, 1);
+                    var backgroundArea = new Rect(0, 0, width, height);
+                    var rectangleGeometry = new RectangleGeometry(backgroundArea);
+                    var rectDrawing = new GeometryDrawing(brush, pen, rectangleGeometry);
+                    group.Children.Add(rectDrawing);
+
+                    var imageArea = new Rect(x + padding, y + padding, image.Width, image.Height);
+                    var imageDrawing = new ImageDrawing(image, imageArea);
+                    group.Children.Add(imageDrawing);
+
+                    // http://stackoverflow.com/questions/6676072/get-system-drawing-bitmap-of-a-wpf-area-using-visualbrush
+                    var visual = new DrawingVisual();
+                    var context = visual.RenderOpen();
+                    context.DrawDrawing(group);
+                    context.Close();
+                    const int dpiScale100 = 96;
+                    var bitmap = new RenderTargetBitmap(
+                        Convert.ToInt32(width),
+                        Convert.ToInt32(height),
+                        dpiScale100,
+                        dpiScale100,
```

Comment:
Is there a reason for hard coding the dpi to 96?

Reply:
This code for plating was added back from Wox. I believe the reason it is set at 96 here is that it is suggested to design WPF app with scale 100% (equivalent to DPI 96) as suggested here: https://stackoverflow.com/questions/4075802/creating-a-dpi-aware-application.